pkg/apis/experimental/v1alpha1/types.go

```diff
@@ -531,13 +535,20 @@ type IngressRule struct {
 	//	  :443 for https.
 	// Both these may change in the future.
 	// Incoming requests are matched against the Host before the IngressRuleValue.
+	// If the Host is unspecified, the Ingress routes all traffic based on the
+	// specified IngressRuleValue.
 	Host string `json:"host,omitempty"`
 	// IngressRuleValue represents a rule to route requests for this IngressRule.
-	IngressRuleValue `json:",inline"`
+	// If unspecified, the rule defaults to a HTTPIngressRuleValue catch-all.
+	// Whether that sends just traffic matching the Host to the default Backend
+	// or all traffic to the default Backend, is left to the controller fulfilling
```

Comment:
Use case for no IngressRuleValue and Host is l4 sni. No way to explain this today since we just have HTTPIngressRuleValue. Alternative is to require a * and another backend; I prefer this hand waving.